extensions/amp-analytics/0.1/events.js

```diff
@@ -438,41 +438,39 @@ export class VideoEventTracker extends EventTracker {
   /** @override */
   add(context, eventType, config, listener) {
     const videoSpec = config['videoSpec'] || {};
-    const endSessionWhenInvisible = videoSpec['end-session-when-invisible'];
-    const excludeAutoplay = videoSpec['exclude-autoplay'];
     const selector = config['selector'] || videoSpec['selector'];
     const selectionMethod = config['selectionMethod'] || null;
-    const on = config['on'];
     const targetReady =
         this.root.getElement(context, selector, selectionMethod);
 
+    const endSessionWhenInvisible = videoSpec['end-session-when-invisible'];
+    const excludeAutoplay = videoSpec['exclude-autoplay'];
+    const on = config['on'];
+
     return this.sessionObservable_.add(event => {
-      const eventData = getData(event);
-      const type = eventData['type'];
-      const details = eventData['details'];
-      const element = user().assertElement(event.target,
-          'No target specified by video session event.');
+      const data = getData(event);
+      const type = data['type'];
+      const details = data['details'];
+      const isVisibleType = type === VideoAnalyticsType.SESSION_VISIBLE;
+      const normalizedType = isVisibleType ? VideoAnalyticsType.SESSION : type;
 
-      // Wait for target selected
-      targetReady.then(analyticsTarget => {
-        if (analyticsTarget.contains(element)) {
-          const isSessionVisible =
-              (type === VideoAnalyticsType.SESSION_VISIBLE);
-
-          // These spec flags filter events that are triggered.
-          // If they apply, we return and do not log the analytics events.
-          const filterSession = (isSessionVisible && !endSessionWhenInvisible);
-          const isAutoplay = details['state'] === PlayingStates.PLAYING_AUTO;
-          const filterAutoplay = (isAutoplay && excludeAutoplay);
-          if (filterSession || filterAutoplay) {
-            return;
-          }
+      if (normalizedType !== on) {
+        return;
+      }
 
-          const mutatedType =
-              isSessionVisible ? VideoAnalyticsType.SESSION : type;
-          if (mutatedType === on) {
-            listener(new AnalyticsEvent(analyticsTarget, mutatedType, details));
-          }
+      if (isVisibleType && !endSessionWhenInvisible) {
+        return;
+      }
+
+      if (excludeAutoplay && details['state'] === PlayingStates.PLAYING_AUTO) {
+        return;
+      }
+
+      const el = dev().assertElement(event.target,
+          'No target specified by video session event.');
+      targetReady.then(target => {
+        if (target.contains(el)) {
```

Comment:
They can be the same, but `Element#contains` is true if you pass the same element.